Backend/FcTypeChecker.hs

```diff
@@ -192,16 +203,16 @@ tcType (FcTyApp ty1 ty2) = do
 tcType (FcTyCon tc) = lookupTyConKindM tc
 
 -- | Type check a list of case alternatives
-tcAlts :: FcType -> [FcAlt 'Fc] -> FcM FcType
+tcAlts :: FcType -> [FcAlt a] -> FcM ([FcAlt 'Fc], FcType)
 tcAlts scr_ty alts
   | null alts = throwError "Case alternatives are empty"
   | otherwise = do
-      rhs_tys <- mapM (tcAlt scr_ty) alts
+      (fc_alts, rhs_tys) <- mapAndUnzipM (tcAlt scr_ty) alts
       ensureIdenticalTypes rhs_tys
       let (ty:_) = rhs_tys
-      return ty
+      return (fc_alts, ty)
 
-tcAlt :: FcType -> FcAlt 'Fc -> FcM FcType
+tcAlt :: FcType -> FcAlt a -> FcM (FcAlt 'Fc, FcType)
 tcAlt scr_ty (FcAlt (FcConPat dc xs) rhs) = case tyConAppMaybe scr_ty of
```

Comment:
In general when something is not implemented yet add the clause with something like `error "not implemented yet"` or something, but do not leave functions in a non-exaustive state (currently `tcAlts` covers only one of the three possible cases).